### PR TITLE
Add metrics to count inconsistencies and fp collisions.

### PR DESCRIPTION
**Pull request by beorn7**
@juliusv @fabxc Just this goody before cutting the RC.

Fixes https://github.com/prometheus/prometheus/issues/691

### CODE REVIEW

File: storage/local/mapper.go [hunk 1]
```diff
@@ -10,6 +10,7 @@ import (
 	"github.com/prometheus/log"
 
 	clientmodel "github.com/prometheus/client_golang/model"
+	"github.com/prometheus/client_golang/prometheus"
```
Comment (juliusv) [inline storage/local/mapper.go]: nit: non-renamed import goes into a separate import group before the renamed one. (of course we could change that rule at some point if it seems too silly)

Comment (beorn7) [inline storage/local/mapper.go]: Yeah, I must check my imports more carefully.

File: storage/local/mapper.go [hunk 2]
```diff
@@ -156,14 +167,24 @@ func (r *fpMapper) maybeAddMapping(
...
+func (m *fpMapper) nextMappedFP() clientmodel.Fingerprint {
+	mappedFP := clientmodel.Fingerprint(atomic.AddUint64((*uint64)(&m.highestMappedFP), 1))
 	if mappedFP > maxMappedFP {
 		panic(fmt.Errorf("more than %v fingerprints mapped in collision detection", maxMappedFP))
 	}
 	return mappedFP
 }
 
+// Describe implements prometheus.Collector.
+func (m *fpMapper) Describe(ch chan<- *prometheus.Desc) {
```
Comment (juliusv) [inline storage/local/mapper.go]: Did you forget to fill out the `Describe` and `Collect` methods?

Comment (beorn7) [inline storage/local/mapper.go]: Dooooooooooooh.

Comment (juliusv) [inline storage/local/mapper.go]: I thought you would have that reaction :)

**Comment by juliusv** [comment]
:+1:

**Pull request merged**